### PR TITLE
Add fuzzy filtering

### PR DESCRIPTION
**Pull request by stuartleeks**
- [X] Add a new fuzzy filter command for fuzzy string matching (e.g.  `srnm` would match `some resource name`)
- [X] Allow setting default key bindings with modifiers. Using `/` for fuzzy filter (compared to `Alt+/` for non-fuzzy filtering)
- [X] Allow uesr-configured key bindings to include modifiers

### CODE REVIEW

File: internal/pkg/keybindings/defaults.go [hunk 1]
```diff
@@ -39,6 +39,7 @@ var DefaultKeys = map[string]interface{}{
 	"commandpanelup":      gocui.KeyArrowUp,
 	"commandpanelenter":   gocui.KeyEnter,
 	"filter":              rune('/'),
```
Comment (lawrencegripper) [inline internal/pkg/keybindings/defaults.go:41]: I wonder if we should take over this one and make fuzzy filter the default

File: internal/pkg/keybindings/global_filter_commandpanel.go [hunk 2]
```diff
@@ -11,16 +11,22 @@ type CommandPanelFilterHandler struct {
 	commandPanelWidget *views.CommandPanelWidget
```
Comment (lawrencegripper) [inline internal/pkg/keybindings/global_filter_commandpanel.go:11]: Classic me typo on file name, thanks 🙏

**Review by lawrencegripper — APPROVED** [review]
Love it 🥰

File: internal/pkg/views/list.go [hunk 3]
```diff
@@ -113,15 +121,44 @@ func (w *ListWidget) itemsToShow() []*expanders.TreeNode {
...
+					// we're done, add any remaining characters
+					if displayIndex+1 < len(displayText) {
+						result += displayText[displayIndex+1:]
+					}
+					break
+				}
+			} else {
+				// not a match - add this character (from the non-lowered display text)
+				result += string(displayText[displayIndex])
+			}
```
Comment (lawrencegripper) [inline internal/pkg/views/list.go:152]: 👆 this is fun!

File: internal/pkg/keybindings/defaults.go (hunk 1 above)
Comment (stuartleeks) [reply internal/pkg/keybindings/defaults.go:41]: Whichever is default, I quite like the idea of `Alt+/` (or `Shift+/`) as the other option 🤷

File: internal/pkg/views/list.go (hunk 3 above)
Comment (stuartleeks) [reply internal/pkg/views/list.go:152]: I'm open to suggestions for improvements. This got a little gnarly as I wanted case-insensitive matching but preserving the case in the output!

File: internal/pkg/keybindings/defaults.go (hunk 1 above)
Comment (stuartleeks) [reply internal/pkg/keybindings/defaults.go]: Fuzzy filter now binds to `/` by default, and the non-fuzzy filter binds to `Alt+/`

File: internal/pkg/views/list.go (hunk 3 above)
Comment (stuartleeks) [reply internal/pkg/views/list.go:151]: I've made a change to the implementation of this that I think simplifies it slightly - let me know what you think :-)

**Review by lawrencegripper — APPROVED** [review]

**Pull request merged**